File: kmp/composeApp/src/commonMain/kotlin/com/bloom/core/models/ExtractedMilestone.kt
package com.bloom.core.models

import com.bloom.core.utils.DatePrecision
import kotlinx.datetime.LocalDateTime

data class ExtractedMilestone(
    val type: String,
    val subType: String? = null,
    val date: LocalDateTime? = null,
    val datePrecision: DatePrecision = DatePrecision.EXACT,
    val rawContent: String,
    val weightKg: Double? = null,
    val heightCm: Double? = null,
)

File: lib/core/models/extracted_milestone.dart
import '../utils/date_precision.dart';

class ExtractedMilestone {
  final String type;
  final String? subType;
  final DateTime? date;
  final DatePrecision datePrecision;
  final String rawContent;
  final String? title;
  final String? location;
  final double? weightKg;
  final double? heightCm;

  const ExtractedMilestone({
    required this.type,
    this.subType,
    this.date,
    this.datePrecision = DatePrecision.exact,
    required this.rawContent,
    this.title,
    this.location,
    this.weightKg,
    this.heightCm,
  });
}
